apps/src/lab2/views/SourcesContainer.tsx

```diff
@@ -56,11 +57,31 @@ const SourcesContainer: React.FC<
     () => getInitialSources(levelProperties, initialSources) || defaultSources
   );
 
+  // Gives an explicit signal to track when we are resetting sources
+  // via level change, a teacher switching between viewing different students, etc.
+  const [sourceInitializationCount, setSourceInitializationCount] = useState(0);
```

Comment:
Out of curiosity, would simply passing in/setting a callback work instead of listening for this state change? Sounds like we really just care about when this happens, not necessarily what the state got set to, right? This might look something like adding a `setReinitializationHandler` or similar on the SourcesContext, and then SketchlabView could do something like `setReinitializationHandler(() => setExcalidrawMountKey(key => key + 1))`